src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AppellantCaseFlagsHandler.java

```diff
@@ -0,0 +1,85 @@
+package uk.gov.hmcts.reform.iacaseapi.domain.handlers.presubmit;
+
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition.APPELLANT_LEVEL_FLAGS;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition.CASE_FLAG_ID;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.CaseFlagDetail;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.CaseFlagValue;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.StrategicCaseFlag;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.StrategicCaseFlagType;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.YesOrNo;
+import uk.gov.hmcts.reform.iacaseapi.domain.handlers.HandlerUtils;
+
+class AppellantCaseFlagsHandler  {
+
+    protected List<CaseFlagDetail> activateCaseFlag(
+        AsylumCase asylumCase,
+        List<CaseFlagDetail> existingCaseFlagDetails,
+        StrategicCaseFlagType caseFlagType,
+        String dateTimeCreated) {
+
+        CaseFlagValue caseFlagValue = CaseFlagValue.builder()
+            .flagCode(caseFlagType.getFlagCode())
+            .name(caseFlagType.getName())
+            .status("Active")
+            .hearingRelevant(YesOrNo.YES)
+            .dateTimeCreated(dateTimeCreated)
+            .build();
+        String caseFlagId = asylumCase.read(CASE_FLAG_ID, String.class).orElse(UUID.randomUUID().toString());
+        List<CaseFlagDetail> caseFlagDetails = existingCaseFlagDetails.isEmpty()
+            ? new ArrayList<>()
+            : new ArrayList<>(existingCaseFlagDetails);
+        caseFlagDetails.add(new CaseFlagDetail(caseFlagId, caseFlagValue));
+
+        return caseFlagDetails;
+    }
+
+    protected List<CaseFlagDetail> deactivateCaseFlag(
+        List<CaseFlagDetail> caseFlagDetails,
+        StrategicCaseFlagType caseFlagType,
+        String dateTimeModified) {
+        if (hasActiveTargetCaseFlag(caseFlagDetails, caseFlagType)) {
```

Comment:
Just for cherry on top I think it would look a bit cleaner and easier to read with below structure:

if (hasActiveTargetCaseFlag(caseFlagDetails, caseFlagType)) {
        return;
  }
    caseFlagDetails = caseFlagDetails.stream().map(detail -> {
                CaseFlagValue value = detail.getCaseFlagValue();
                if (!isActiveTargetCaseFlag(value, caseFlagType)) {
                    return detail;
                } 
// also probably worth extracting below builder to a separate method
              return new CaseFlagDetail(detail.getId(), CaseFlagValue.builder()
                        .flagCode(value.getFlagCode())
                        .name(value.getName())
                        .status("Inactive")
                        .hearingRelevant(value.getHearingRelevant())
                        .dateTimeModified(dateTimeModified)
                        .build());
            }).collect(Collectors.toList());